### PR TITLE
Adding possibility to send body null and tests

### PR DESCRIPTION
**Pull request by alallema**
**Description**
Adding the possibility for the client to send body null.
Fixing update settings method.
Adding tests to check if nil could be sent properly.

**Issue related**
#121

****
Still not working on `update_synonyms()` and `update_stop_words()` method because the meilisearch API doesn't seem to take null as arguments

### CODE REVIEW

File: spec/meilisearch/index/settings_spec.rb
Comment (curquiza) [inline spec/meilisearch/index/settings_spec.rb:397]:
```suggestion
    it 'updates attributes for faceting at null' do
```

**Review by curquiza — APPROVED** [review]
I confirm, the synonyms and the stop words cannot be updated with `null` via the MeiliSearch API. I will open an issue to fix this in transplant once all the settings are implemented in.

Waiting for the approval of @bidoubiwa before any merge

**Comment by curquiza** [comment]
2 remarks:
- your issue was not linked with this PR, even if you mentioned it in the description. I added it 😉 It's important so that the issue is closed automatically by GitHub once the PR is merged. Sometimes (depending on how you mention it and how many issues you mention) GitHub does not link the issue in the PR, unfortunately you have to check every time 😅
- The linter fails: this section in the CONTRIBUTING might help you fix it: https://github.com/meilisearch/meilisearch-ruby/blob/main/CONTRIBUTING.md#linter

**Review by curquiza — APPROVED** [review]
🚀

**Review by bidoubiwa — APPROVED** [review]
🔥🔥🔥

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**
